### ISSUE TITLE
Migrate from psycopg2 to psycopg(e) for postgres

### ISSUE THREAD

**Issue by logan-hcg** [TobikoData/sqlmesh#3451]
While not pressing (as psycopg2 continues to be supported), I think most new effort is behind the psycopg3 implementation: https://www.psycopg.org/install/

**Comment by AliRaza5959** [comment]
yes, it would help move towards the binary rather than build from source for psycopg2.
